collectors/collector_manager.go

```diff
@@ -24,21 +24,25 @@ type CollectorManager struct {
 	scrapeDurations *prometheus.SummaryVec
 }
 
-func NewCollectorManager(endpoint string) *CollectorManager {
-	client := logstashclient.NewClient(endpoint)
-
-	collectors := getCollectors(client)
+func NewCollectorManager(endpoints []string) *CollectorManager {
+	collectors := getCollectors(endpoints)
 
 	scrapeDurations := getScrapeDurationsCollector()
 	prometheus.MustRegister(version.NewCollector("logstash_exporter"))
 
 	return &CollectorManager{collectors: collectors, scrapeDurations: scrapeDurations}
 }
 
-func getCollectors(client logstashclient.Client) map[string]Collector {
+func getCollectors(endpoints []string) map[string]Collector {
 	collectors := make(map[string]Collector)
-	collectors["nodeinfo"] = nodeinfo.NewNodeinfoCollector(client)
-	collectors["nodestats"] = nodestats.NewNodestatsCollector(client)
+	for _, endpoint := range endpoints {
+		client := logstashclient.NewClient(endpoint)
+
+		// NOTE: This could be structured better, but this avoids adding another level of map depth
+		collectors[endpoint + "_nodeinfo"] = nodeinfo.NewNodeinfoCollector(client)
+		collectors[endpoint + "_nodestats"] = nodestats.NewNodestatsCollector(client)
+	}
+	
```

Comment:
The above code is what I like the least here. To avoid changing the collector object schema, I am prefixing the map keys with the endpoint (IP:Port) string. I'm not sure if/how this translates to prometheus metrics outputs, but I welcome feedback about how to make this more elegant. It's admittedly shoehorned in for now.

Reply:
The actual replica/pod name is available through both the Kubernetes client API, and through Logstash's own `_node/stats` (`host` and `name` are the pod name). Moreover, `ServiceEndpoint.labels` might also be useful to populate a collector name, if we passed the whole struct in to `NewCollectorManager`, rather than reducing it to an endpoint string in `main()`.